src/Rules/RuleLevelHelper.php

```diff
@@ -48,6 +49,16 @@ public function isThis(Expr $expression): bool
 	/** @api */
 	public function accepts(Type $acceptingType, Type $acceptedType, bool $strictTypes): bool
 	{
+		$checkForUnion = $this->checkUnionTypes;
+
+		if (
+			$this->checkBenevolentUnionTypes
+			&& $acceptedType instanceof BenevolentUnionType
```

Comment:
This is too naive - will not see nested BenevolentUnionType. Needs similar logic as `checkExplicitMixed` and use `TypeTraverser`.

Reply:
I have a test for (\DateTimeImmutable|false) and for array<(\DateTimeImmutable|false)>.

What do you have in mind with "nested BenevolentUnionType", I'll add new tests.

Reply:
I guess things like `(float|int)|string`